src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1724,104 +1726,294 @@ private BoundExpression BindAssignment(AssignmentExpressionSyntax node, Diagnost
             if (!diagnostics.HasAnyErrors())
             {
                 diagnostics.AddRange(bag);
             }
 
             if ((object)deconstructMethod == null)
             {
-                return new BoundDeconstructionAssignmentOperator(node, checkedVariables, boundRHS, ErrorMethodSymbol.UnknownMethod,
-                                                                ImmutableArray<BoundDeconstructionAssignmentOperator.AssignmentInfo>.Empty,
-                                                                ErrorTypeSymbol.UnknownResultType,
-                                                                hasErrors: true);
+                return null;
             }
 
-            // figure out the pairwise conversions
             var deconstructParameters = deconstructMethod.Parameters;
-            var assignments = checkedVariables.SelectAsArray((variable, index, parameters) => MakeAssignmentInfo(variable, parameters[index].Type, node, diagnostics), deconstructParameters);
 
-            TypeSymbol voidType = GetSpecialType(SpecialType.System_Void, diagnostics, node);
-            return new BoundDeconstructionAssignmentOperator(node, checkedVariables, boundRHS, deconstructMethod, assignments, voidType);
+            var deconstructionStep = new DeconstructStep()
+            {
+                DeconstructMemberOpt = deconstructMethod,
+                TargetPlaceholder = targetPlaceholder,
+                OutputPlaceholders = deconstructParameters.SelectAsArray((p, s) => new BoundDeconstructValuePlaceholder(s, p.Type), syntax)
+            };
+
+            return deconstructionStep;
+        }
+
+        /// <summary>
+        /// Takes the outputs from the previous deconstructionStep and depending on the structure of variables, will generate further deconstructions, or simply assignments.
+        /// </summary>
+        private bool DeconstructOrAssignOutputs(
+                        DeconstructStep deconstructionStep,
+                        ImmutableArray<BoundExpression> variables,
+                        AssignmentExpressionSyntax syntax,
+                        DiagnosticBag diagnostics,
+                        ArrayBuilder<DeconstructStep> deconstructionSteps,
+                        ArrayBuilder<AssignmentStep> assignmentSteps)
+        {
+            for (int i = 0; i < variables.Length; i++)
+            {
+                var variable = variables[i];
+                var valuePlaceholder = deconstructionStep.OutputPlaceholders[i];
+
+                if (variable.Kind == BoundKind.DeconstructionVariables)
+                {
+                    var nestedVariables = ((BoundDeconstructionVariables)variable).Variables;
+
+                    if (!DeconstructIntoSteps(valuePlaceholder, syntax, diagnostics, nestedVariables, deconstructionSteps, assignmentSteps))
+                    {
+                        return false;
```

Comment:
Done. I'm now collecting all the deconstruction steps that I can, even after encountering an error.